eng/tools/typespec-validation/test/folder-structure.test.ts

```diff
@@ -0,0 +1,157 @@
+import { FolderStructureRule } from "../src/rules/folder-structure.js";
+import { TsvTestHost } from "./tsv-test-host.js";
+import { strict as assert } from "node:assert";
+
+describe("folder-structure", function () {
+  it("should fail if no tspconfig has incorrect extension", async function () {
+    let host = new TsvTestHost();
+    host.globby = async () => {
+      return ["/foo/bar/tspconfig.yml"];
+    };
+
+    const result = await new FolderStructureRule().execute(host, TsvTestHost.folder);
+    assert(result.errorOutput);
+    assert(result.errorOutput.includes("Invalid config file"));
+  });
+
+  it("should fail if no folder under specification/ is capitalized", async function () {
+    let host = new TsvTestHost();
+    host.globby = async () => {
+      return ["/foo/bar/tspconfig.yaml"];
+    };
+    host.normalizePath = () => {
+      return "/gitroot";
+    };
+
+    const result = await new FolderStructureRule().execute(host, "/gitroot/specification/Foo/Foo");
+    assert(result.errorOutput);
+    assert(result.errorOutput.includes("must be lower case"));
+  });
+
+  it("should fail if package folder is more than 3 levels deep", async function () {
+    let host = new TsvTestHost();
+    host.globby = async () => {
+      return ["/foo/bar/tspconfig.yaml"];
+    };
+    host.normalizePath = () => {
+      return "/gitroot";
+    };
+
+    const result = await new FolderStructureRule().execute(
+      host,
+      "/gitroot/specification/Foo/Foo/Foo/Foo",
```

Comment:
```suggestion
      "/gitroot/specification/foo/Foo/Foo/Foo",
```

I think the service-family folder should be lower-case, unless we are trying to make it fail?